packages/state-manager/src/sagas/messages/verifyMessage/updateMessagesVerification.saga.ts

```diff
@@ -0,0 +1,26 @@
+import { select, spawn } from 'typed-redux-saga'
+import { ChannelMessage } from '../../publicChannels/publicChannels.types'
+import { messagesSelectors } from '../messages.selectors'
+import { verifyMessage } from './verifyMessages.saga'
+import { getCrypto } from 'pkijs'
+
+export function* updateMessagesVerificationSaga(): Generator {
+  const crypto = getCrypto()
+  const statuses = yield* select(messagesSelectors.messagesVerificationStatus)
+  console.log('statuses', statuses)
+
+  const allMessages = yield* select(messagesSelectors.currentPublicChannelMessagesBase)
```

Comment:
You're running crypto method (`verifyMessage`) for each channel message we store in redux persist even though many of them are already verified. This is **super expensive** and it's definitely not something we want to do as we'll probably shortly be running into performance issues

Reply:
Ok, I  see that few lines below you're filtering out already verified messages but it happens outside the selector so we're not getting advantage of memoization and all those entries goes through the loop which is bad

Reply:
i think you are wrong, I'm running this method only for non-existent messages in messagesVerificationStatus (not for those with the verify value true or false)